ctru-rs/src/gfx.rs

```diff
@@ -9,17 +10,40 @@ use crate::services::gspgpu::{self, FramebufferFormat};
 /// provides helper functions and utilities for software rendering.
 ///
 /// The service exits when this struct is dropped.
-pub struct Gfx(());
+pub struct Gfx {
+    top_screen: RefCell<TopScreen>,
+    bottom_screen: RefCell<BottomScreen>,
+}
 
-/// Available screens on the 3DS
-#[derive(Copy, Clone, Debug)]
-pub enum Screen {
-    /// The top screen
-    Top,
-    /// The bottom screen
-    Bottom,
+/// Trait implemented by TopScreen and BottomScreen for common methods
+pub trait Screen {
+    /// Returns the libctru value for the Screen kind
+    fn into_raw(&self) -> ctru_sys::gfxScreen_t;
```

Comment:
This should be `as_raw` since it's "free" and doesn't take ownership:
```suggestion
    fn as_raw(&self) -> ctru_sys::gfxScreen_t;
```

More info about naming guidelines:
https://rust-lang.github.io/api-guidelines/naming.html